code/modules/mob/living/simple_animal/vore/otie.dm

```diff
@@ -0,0 +1,297 @@
+	meat_amount = 5
+
+	old_x = -16
+	old_y = 0
+	pixel_x = -16
+	pixel_y = 0
+
+	var/mob/living/carbon/human/friend
+	var/tamed = 0
+	var/threat = null
+
+/mob/living/simple_animal/otie/fluff
+	name = "otie"
+	desc = "The classic bioengineered longdog. This one might even tolerate you!"
+	icon_state = "otie"
+	icon_living = "otie"
+	icon_rest = "otie_rest"
+	faction = "neutral"
+
+/mob/living/simple_animal/otie/fluff/cotie
+	name = "tamed otie"
+	desc = "The classic bioengineered longdog. This one has a nice little collar on its neck. However a proper domesticated otie is an oxymoron and the collar is likely just a decoration."
+	icon_state = "cotie"
+	icon_living = "cotie"
+	icon_rest = "cotie_rest"
+	faction = "neutral"
+
+/mob/living/simple_animal/otie/fluff/security
+	name = "guard otie"
+	desc = "The V.A.R.M.A.corp bioengineering division flagship product on trained optimal snowflake guard dogs."
+	icon_state = "sotie"
+	icon_living = "sotie"
+	icon_rest = "sotie_rest"
+	icon_dead = "sotie-dead"
+	faction = "neutral"
+	tamed = 1
+
+	var/check_records = 1 // If true, arrests people without a record.
+	var/check_arrest = 1 // If true, arrests people who are set to arrest.
+
+//Holy shit oh god how do I make it not destroy and devour everything indiscriminately!
+
+/mob/living/simple_animal/otie/PunchTarget()
+	if(istype(target_mob,/mob/living/simple_animal/mouse))
+		return EatTarget()
+	else ..()
+
+/mob/living/simple_animal/otie/Found(var/atom/found_atom)
+	if(!SA_attackable(found_atom))
+		return null
+	if(istype(found_atom,/mob/living/simple_animal/mouse))
+		return found_atom
+	if(will_eat(found_atom))
+		if(found_atom in friends)
+			return null
+		else if(found_atom in faction_friends)
+			return null
+		else if (friend == found_atom)
+			return null
+		else if(tamed == 1 && ishuman(found_atom))
+			return null
+		else if(tamed == 1 && isrobot(found_atom))
+			return null
+		else
+			return found_atom
+
+/mob/living/simple_animal/otie/fluff/security/Found(var/atom/found_atom)
+	if(!SA_attackable(found_atom))
+		return null
+	if(istype(found_atom,/mob/living/simple_animal/mouse))
+		return found_atom
+	if(check_threat(found_atom) >= 4)
+		return found_atom
+	if(will_eat(found_atom))
+		if(found_atom in friends)
+			return null
+		else if(found_atom in faction_friends)
+			return null
+		else if (friend == found_atom)
+			return null
+		else if(tamed == 1 && ishuman(found_atom))
+			return null
+		else if(tamed == 1 && isrobot(found_atom))
+			return null
+		else
+			return found_atom
+
+/mob/living/simple_animal/otie/fluff/security/proc/check_threat(var/mob/living/M)
+	if(!M || !ishuman(M) || M.stat == DEAD || src == M)
+		return 0
+	return M.assess_perp(0, 0, 0, check_records, check_arrest)
+
+/mob/living/simple_animal/otie/FindTarget()
+	var/atom/T = null
+	for(var/atom/A in ListTargets(view_range))
+
+		if(A == src)
+			continue
+
+		var/atom/F = Found(A)
+		if(F)
+			T = F
+			break
+		else if(specific_targets)
+			return 0
```

Comment:
You have specific_targets set to true on all of these, so really you could handle your tamed checks in Found() and not override this one here. On the one that does the sechud broadcast, you could override that in set_target instead.

The reason I suggest doing that instead is because this is a big proc with lots of little fiddly bits that may be changed in the future, and you wouldn't get those changes and your mob may stop behaving like one might expect. set_target is less likely to change since it's a very simple proc... there's not really much to fix about it, and Found() will never be changed because it's whole purpose is to be overridden by children.